bowtypes.go

```diff
@@ -82,20 +75,28 @@ func (t Type) Convert(i interface{}) interface{} {
 // IsSupported ensures that the type is currently supported by Bow
 // and match a convertible concrete type.
 func (t Type) IsSupported() bool {
-	_, ok := mapTypeArrowDataType[t]
+	_, ok := mapBowToArrowTypes[t]
 	return ok
 }
 
 func (t Type) String() string {
-	at, ok := mapTypeArrowDataType[t]
+	at, ok := mapBowToArrowTypes[t]
 	if !ok {
 		return "undefined"
 	}
 	return at.Name()
 }
 
-func getTypeFromArrowType(arrowType arrow.DataType) Type {
-	t, ok := mapArrowTTypes[arrowType]
+func getBowTypeFromArrowName(arrowName string) Type {
+	t, ok := mapArrowNameToBowTypes[arrowName]
+	if !ok {
+		return Unknown
+	}
+	return t
+}
+
+func getBowTypeFromArrowType(arrowType arrow.DataType) Type {
+	t, ok := mapArrowToBowTypes[arrowType]
```

Comment:
from 't' to 'typ' to be consistent with your changes
I can't suggest easy apply change as the return statement is not part of the MR it can't be included